### ISSUE TITLE
(docs) Consider moving/removing fragments section of GraphQL API

### ISSUE THREAD

**Issue by tesseralis** [gatsbyjs/gatsby#21109]
## Summary

Consider moving the "Fragments" section of the [GraphQL API](https://www.gatsbyjs.org/docs/graphql-api/#query-fragments) to its own page(s). Alternately, remove these sections, since their content is already covered under the [Gatsby Image API](https://www.gatsbyjs.org/docs/gatsby-image/#image-query-fragments).

### Motivation

The "Fragments" section contains GraphQL fragments for two plugins: Contentful and ImageSharp. It makes them hard to find when they're hidden at the bottom of the GraphQL API.

Additionally, this requires us to pull in a one-time use MDX component, `<GraphQLApiQuery>`, which [can interfere with translated docs](https://github.com/gatsbyjs/gatsby/issues/20795). There are other work-arounds for this, like making the component global 

Furthermore, the static-query used is included in the main site bundle for some reason, resulting in significant bloat for the site (#21470). 10% of our main website bundle comes from these queries! Moving them to page queries, or removing them alltogether, alleviates this.

## Steps to resolve this issue

* Create a new overview page "GraphQL Fragments" that introduces the concept of fragments and links to the two pages below.
* Create two new pages "Contentful Fragments" and "ImageSharp Fragments", each pulling in the APIs for the specific plugin using the new `template-api` defined in #20763 

# Related Issues

Part of: #20795
Depends on: #20763
